examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2105,623 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
+					"Japan Airlines",
+					"LATAM Chile",
+					"LATAM Brasil",
+					"Malasya Airlines",
+					"Canadian Airlines",
+				}},
+			},
+		}
+
+		salesResult, salesErr := salesColl.InsertMany(ctx, sales)
+		airlinesResult, airlinesErr := airlinesColl.InsertMany(ctx, airlines)
+		airAlliancesResult, airAlliancesErr := airAlliancesColl.InsertMany(ctx, airAlliances)
+
+		// End Aggregation Example 1
+
+		require.NoError(t, salesErr)
+		require.Len(t, salesResult.InsertedIDs, 6)
+		require.NoError(t, airlinesErr)
+		require.Len(t, airlinesResult.InsertedIDs, 7)
+		require.NoError(t, airAlliancesErr)
+		require.Len(t, airAlliancesResult.InsertedIDs, 3)
+	}
+	{
+		// Start Aggregation Example 2
+		pipeline := mongo.Pipeline{
+			{
+				{"$match", bson.D{
+					{"items.fruit", "banana"},
+				}},
+			},
+			{
+				{"$sort", bson.D{
+					{"date", 1},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 2
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+		requireCursorLength(t, cursor, 5)
+	}
+	{
+		// Start Aggregation Example 3
+		pipeline := mongo.Pipeline{
+			{
+				{"$unwind", "$items"},
+			},
+			{
+				{"$match", bson.D{
+					{"items.fruit", "banana"},
+				}},
+			},
+			{
+				{"$group", bson.D{
+					{"_id", bson.D{
+						{"day", bson.D{
+							{"$dayOfWeek", "$date"},
+						}},
+					}},
+					{"count", bson.D{
+						{"$sum", "$items.quantity"},
+					}},
+				}},
+			},
+			{
+				{"$project", bson.D{
+					{"dayOfWeek", "$_id.day"},
+					{"numberSold", "$count"},
+					{"_id", 0},
+				}},
+			},
+			{
+				{"$sort", bson.D{
+					{"numberSold", 1},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 3
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+		requireCursorLength(t, cursor, 4)
+	}
+	{
+		// Start Aggregation Example 4
+		pipeline := mongo.Pipeline{
+			{
+				{"$unwind", "$items"},
+			},
+			{
+				{"$group", bson.D{
+					{"_id", bson.D{
+						{"day", bson.D{
+							{"$dayOfWeek", "$date"},
+						}},
+					}},
+					{"items_sold", bson.D{
+						{"$sum", "$items.quantity"},
+					}},
+					{"revenue", bson.D{
+						{"$sum", bson.D{
+							{"$multiply", bson.A{"$items.quantity", "$items.price"}},
+						}},
+					}},
+				}},
+			},
+			{
+				{"$project", bson.D{
+					{"day", "$_id.day"},
+					{"revenue", 1},
+					{"items_sold", 1},
+					{"discount", bson.D{
+						{"$cond", bson.D{
+							{"if", bson.D{
+								{"$lte", bson.A{"$revenue", 250}},
+							}},
+							{"then", 25},
+							{"else", 0},
+						}},
+					}},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 4
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+		requireCursorLength(t, cursor, 4)
+	}
+	{
+		// Start Aggregation Example 5
+		pipeline := mongo.Pipeline{
+			{
+				{"$lookup", bson.D{
+					{"from", "air_airlines"},
+					{"let", bson.D{
```

Comment:
It looks like `let` in `$lookup` was introduced in MongoDB 3.6.
https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/#join-conditions-and-uncorrelated-sub-queries has the note "New in version 3.6."